pulser/pulse.py

```diff
@@ -64,9 +64,10 @@ def __new__(cls, *args, **kwargs):
         else:
             return object.__new__(cls)
 
-    def __init__(self, amplitude: Waveform, detuning: Waveform,
+    def __init__(self, amplitude: Union[Parametrized, Waveform],
+                 detuning: Union[Parametrized, Waveform],
```

Comment:
Nit: Let's keep `Parametrized` as the second parameter, just for consistency's sake.